subsys/net/ip/dhcpv4.c

```diff
@@ -150,15 +150,22 @@ static inline bool dhcpv4_add_sname(struct net_pkt *pkt)
 /* Create DHCPv4 message and add options as per message type */
 static struct net_pkt *dhcpv4_create_message(struct net_if *iface, u8_t type,
 					     const struct in_addr *ciaddr,
+					     const struct in_addr *src,
 					     const struct in_addr *server_addr,
 					     bool server_id, bool requested_ip)
 {
 	NET_PKT_DATA_ACCESS_DEFINE(dhcp_access, struct dhcp_msg);
-	const struct in_addr src = INADDR_ANY_INIT;
+	const struct in_addr any = INADDR_ANY_INIT;
+	const struct in_addr *addr;
 	size_t size = DHCPV4_MESSAGE_SIZE;
 	struct net_pkt *pkt;
 	struct dhcp_msg *msg;
 
+	if (src == NULL)
```

Comment:
Missing {}

Reply:
Oh yes. Someone needs to file a bug against checkpatch as this is not detected automatically...!